runtime/v2/shim/util_unix.go

```diff
@@ -63,20 +66,20 @@ func AdjustOOMScore(pid int) error {
 	return nil
 }
 
-// SocketAddress returns an abstract socket address
+// SocketAddress returns a socket address
 func SocketAddress(ctx context.Context, id string) (string, error) {
 	ns, err := namespaces.NamespaceRequired(ctx)
 	if err != nil {
 		return "", err
 	}
 	d := sha256.Sum256([]byte(filepath.Join(ns, id)))
-	return filepath.Join(string(filepath.Separator), "containerd-shim", fmt.Sprintf("%x.sock", d)), nil
+	return filepath.Join("/run/containerd/s", fmt.Sprintf("%x", d)), nil
 }
 
 // AnonDialer returns a dialer for an abstract socket
 func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
 	address = strings.TrimPrefix(address, "unix://")
-	return dialer.Dialer("\x00"+address, timeout)
+	return dialer.Dialer(address, timeout)
```

Comment:
This change may break third party projects importing this library, e.g. gvisor shim and kata shim.

So file-based sockets should be prefixed with "file://". An unprefixed strings should still be  interpreted as an abstract socket path.

Reply:
I think that could work.  Instead of file what about `unix://`? Most of our code strips `unix://` anyways.

Reply:
Either is fine to me as long as it is well documented, but we already have `file://` here: https://github.com/containerd/containerd/blob/7868e8d6aab20a44dafe6f330aa8e2afadf3b750/pkg/process/io.go#L112